pyfstat/core.py

```diff
@@ -224,9 +224,12 @@ def predict_fstat(
 class BaseSearchClass(object):
     """ The base search class providing parent methods to other searches """
 
-    def _add_log_file(self):
+    def _add_log_file(self, header=[]):
```

Comment:
Thanks, I wasn't aware of that weird scoping issue! It seems here there's no problem, since the header list is not actually modified inside the function. (I think you're referring to the "if the function modifies the object (e.g. by appending an item to a list), the default value is in effect modified" point in the docs, right?) But it might still be good to implement your suggestion, and I can later audit the rest of the code for possible harmful instances of this type of issue.

Reply:
Yes, that's what I mean. It shouldn't be an issue in this context, but I think is worth fixing, given that no code relies on this yet.

Reply:
Can you put it here as a code suggestion? I'd like to see how that feature over here compares to the one from gitlab.

(Let me know if you need to be added as a project member to do so.)

Reply:
I've tried to add it myself, but it seems I have to be a member

Reply:
I've asked Greg to add you, but have committed the change directly in the meantime. Thanks!